mono/metadata/object.c

```diff
@@ -4010,38 +4010,55 @@ static int num_main_args = 0;
 MonoArray*
 mono_runtime_get_main_args (void)
 {
+	HANDLE_FUNCTION_ENTER ();
 	MONO_REQ_GC_UNSAFE_MODE;
 	MonoError error;
-	MonoArray *result = mono_runtime_get_main_args_checked (&error);
-	mono_error_assert_ok (&error);
-	return result;
+	MonoArrayHandle arg_array = mono_runtime_get_main_args_handle (&error);
+	if (!is_ok (&error))
+		return NULL;
```

Comment:
functions wrapped in `HANDLE_FUNCTION_{ENTER, LEAVE}` should have a single return.  I suggest:

```c
    MonoArray *result = NULL;
    MonoArrayHandle arg_arrray = mono_runtime_get_main_args_handle (&error);
    goto_if_nok (&error, leave);
    result = MONO_HANDLE_RAW (arg_array);

leave:
    HANDLE_FUNCTION_RETURN_VAL(result);
```